patroni/__main__.py

```diff
@@ -88,28 +89,21 @@ def load_dynamic_configuration(self) -> None:
         .. note::
             This method is called only once, at the time when Patroni is started.
         """
-        from patroni.exceptions import DCSError
-        while True:
-            try:
-                cluster = self.dcs.get_cluster()
-                if cluster and cluster.config and cluster.config.data:
-                    if self.config.set_dynamic_configuration(cluster.config):
-                        self.dcs.reload_config(self.config)
-                        self.watchdog.reload_config(self.config)
-                elif not self.config.dynamic_configuration and 'bootstrap' in self.config:
-                    if self.config.set_dynamic_configuration(self.config['bootstrap']['dcs']):
-                        self.dcs.reload_config(self.config)
-                        self.watchdog.reload_config(self.config)
-                break
-            except DCSError:
-                logger.warning('Can not get cluster from dcs')
-                time.sleep(5)
+        cluster = self.dcs.cluster
```

Comment:
It is better explicitly pass `cluster` object as an argument instead of using cached value, because the last one may expire.

Reply:
Also, probably this method should be renamed to `apply_dynamic_configuration`.